### PR TITLE
Refactor and migrate documentation 

### LINKED ISSUE #665: Set up project for product docs
#### OpenStack Release
all

#### Description
Set up [product docs](https://docs.f5net.com/display/PDBLDDO/Product+docs) for this repo.

### PR DESCRIPTION
**Pull request by jputrino**
@rsalsa @richbrowne 

#### What issues does this address?
Fixes #665 - product and feature docs; #805 

#### What's this change do?
- sets up product docs (all agent configs are now documented in this repo)
- add docker makefile commands and preview makefile command for building docs locally in docker container
- update requirements, conf.py
- set up travis deployment to aws
- update sphinx project configs

#### Where should the reviewer start?
View the documentation at the staging site (URL shared offline).

### CODE REVIEW

File: docs/README.rst [hunk 1]
```diff
@@ -0,0 +1,733 @@
...
+                                                            via the oslo ``tunnel_sync``
+                                                            message queues. The agent
+                                                            registers BIG-IP devices as tunnel
+                                                            peers based on this setting.
+
+                                                            This setting must be the same on
+                                                            all OpenStack nodes (controller,
+                                                            compute, and network).
+------------------------------------ ---------------------- ----------------------------------- ------------------------------------------- ---------------------
+f5_populate_static_arp               boolean                Controls BIG-IP                     TRUE: the agent adds static entries         TRUE
```
Comment (richbrowne) [inline docs/README.rst]: This is False in the configuration file.  It should be changed to false.

File: docs/conf.py [hunk 2]
```diff
@@ -349,6 +363,33 @@
 .. |f5_icontrol_rpm_url| replace:: https://github.com/F5Networks/f5-icontrol-rest-python/releases/download/v%(f5_icontrol_version)s/f5-icontrol-rest-%(f5_icontrol_version)s-1.el7.noarch.rpm
 .. |f5_icontrol_rpm_package| replace:: f5-icontrol-rest-%(f5_icontrol_version)s-1.el7.noarch.rpm
 .. |f5_icontrol_deb_package| replace:: python-f5-icontrol-rest_%(f5_icontrol_version)s-1_1404_all.deb
+.. |agent-long| replace:: F5 Agent Plugin for OpenStack Neutron
```
Comment (richbrowne) [inline docs/conf.py]: Maybe F5 LBaaS Agent for OpenStack Neutron

File: docs/l2-adjacent-mode.rst [hunk 3]
```diff
@@ -0,0 +1,365 @@
...
+.. todo:: add Cisco APIC/ACI deployment solution guide and link to it here
+
+.. important::
+
+   The |agent-short| L2/L3 segmentation mode settings must match the configurations of your existing external network and BIG-IP device(s).
+
+Next steps
+----------
+
+- If this is your initial launch, :ref:`start the F5 BIG-IP Controller for OpenStack <topic-start-the-agent>`.
```
Comment (richbrowne) [inline docs/l2-adjacent-mode.rst]: Agent

File: .github/ISSUE_TEMPLATE.md [hunk 4]
```diff
@@ -22,10 +22,10 @@
...
 #### Description
-<Describe the bug in detail, steps taken prior to encountering the issue, yand a short explanation of you have deployed openstack and F5® agent>
+<Describe the bug in detail, steps taken prior to encountering the issue, yand a short explanation of you have deployed openstack and F5 agent>
 
 #### Deployment
-<Explain in reasonable detail your OpenStack deployment, the F5® OpenStack agent, and BIG-IP®(s)>
-<Example: Single OpenStack controller with one F5® agent managing a cluster of 4 BIG-IP® VEs>
-<Example: Three OpenStack controllers in HA, each with one standalone F5® agent managing a single BIG-IP® appliance>
+<Explain in reasonable detail your OpenStack deployment, the F5 OpenStack agent, and BIG-IP(s)>
+<Example: Single OpenStack controller with one F5 agent managing a cluster of 4 BIG-IP VEs>
```
Comment (richbrowne) [inline .github/ISSUE_TEMPLATE.md]: Agents

**Comment by jputrino** [comment]
@richbrowne I think we're good to go now.

**Comment by richbrowne** [comment]
Ok, I will look at this after our OpenStack meeting.

**Pull request merged**